Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```
@@ -0,0 +1,355 @@
+        documentation: template.Body,
+      };
+      completionList.push(item);
+    });
+
+    buildInfunctionsMap.forEach((value, key) => {
+      const item = {
+        label: key,
+        kind: CompletionItemKind.Function,
+        insertText: key + '(' + this.removeParamFormat(value.Params.toString()) + ')',
+        documentation: value.Introduction,
+      };
+      completionList.push(item);
+    });
+
+    const matchResult = this.matchedStates(params);
+    // TODO: more precise match
+    if (
+      matchResult &&
+      matchResult.matched &&
+      matchResult.state &&
+      allowedCompletionStates.includes(matchResult.state.toLowerCase())
+    ) {
+      return Promise.resolve({ isIncomplete: true, items: completionList });
+    } else {
+      return Promise.resolve(null);
+    }
+  }
+
+  protected validate(document: TextDocument): void {
+    this.cleanPendingValidation(document);
+    setTimeout(() => {
+      this.pendingValidationRequests.delete(document.uri);
+      this.doValidate(document);
+    });
+  }
+
+  protected cleanPendingValidation(document: TextDocument): void {
+    const request = this.pendingValidationRequests.get(document.uri);
+    if (request !== undefined) {
+      clearTimeout(request);
+      this.pendingValidationRequests.delete(document.uri);
+    }
+  }
+
+  protected doValidate(document: TextDocument): void {
+    let text = document.getText();
+    let lineOffset = 0;
+    const LGDocument = this.getLGDocument(document);
+    if (LGDocument) {
+      // concat new content for validate
+      const { content, template } = LGDocument;
+      const updatedTemplate = {
+        Name: template.Name,
+        Parameters: template.Parameters,
+        Body: text,
+      };
+      try {
+        text = updateTemplateInContent(content, updatedTemplate);
+        lineOffset = getTemplatePositionOffset(content, updatedTemplate);
+        // eslint-disable-next-line no-empty
+      } catch (error) {}
```

Please put a comment on why silencing errors. It give next developers some insights on what actually happened.